app/script/storage/StorageService.js

```diff
@@ -86,6 +89,22 @@ z.storage.StorageService = class StorageService {
     });
   }
 
+  _setAfterDbInitCallback(callback) {
+    this._afterDbInit = callback;
+  }
+
+  // Hooks
+  addUpdatedListener(storeName, callback) {
+    if (!this.db) {
+      // waiting for the DB to be initialized
+      return this._setAfterDbInitCallback(() => this.addUpdatedListener(storeName, callback));
+    }
+    this.db[storeName].hook('updating', function(modifications, primaryKey, obj, transation) {
```

Comment:
Shall we use an arrow function here?

Reply:
No we cannot because dexie wants us to use `this` inside.
If we use arrow, then we have the lexical `this`, not the one from `dexie`